### ISSUE TITLE
ErrorResponseProcessor in a Groovy project throws CNF exception

### ISSUE THREAD

**Issue by timyates** [micronaut-projects/micronaut-core#8184]
### Expected Behavior

Given a custom exception `WhoopsException`, I should be able to create an `ExceptionHandler` which processes a response for this via `ErrorResponseProcessor`, ie:

```
@Produces
@Singleton
class MyExceptionHandler implements ExceptionHandler<WhoopsException, HttpResponse<?>>{

    private final ErrorResponseProcessor<?> errorResponseProcessor

    MyExceptionHandler(ErrorResponseProcessor<?> errorResponseProcessor) {
        this.errorResponseProcessor = errorResponseProcessor
    }

    @Override
    HttpResponse<?> handle(HttpRequest request, WhoopsException exception) {
        ErrorContext errorContext = ErrorContext.builder(request)
                .cause(exception)
                .errorMessage(exception.getMessage())
                .build()
        errorResponseProcessor.processResponse(errorContext, unprocessableEntity())
    }
}
```

### Actual Behaviour

However, when it attempts to call `processResponse`, it fails with

```
java.lang.NoClassDefFoundError: io/micronaut/jackson/JacksonConfiguration
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3137)
	at java.base/java.lang.Class.getDeclaredConstructors(Class.java:2357)
	at org.codehaus.groovy.reflection.CachedClass$2.lambda$initValue$4(CachedClass.java:71)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at org.codehaus.groovy.reflection.CachedClass$2.initValue(CachedClass.java:76)
	at org.codehaus.groovy.reflection.CachedClass$2.initValue(CachedClass.java:66)
	at org.codehaus.groovy.util.LazyReference.getLocked(LazyReference.java:50)
	at org.codehaus.groovy.util.LazyReference.get(LazyReference.java:37)
	at org.codehaus.groovy.reflection.CachedClass.getConstructors(CachedClass.java:259)
	at groovy.lang.MetaClassImpl.<init>(MetaClassImpl.java:223)
	at groovy.lang.MetaClassImpl.<init>(MetaClassImpl.java:233)
	at groovy.lang.MetaClassRegistry$MetaClassCreationHandle.createNormalMetaClass(MetaClassRegistry.java:166)
	at groovy.lang.MetaClassRegistry$MetaClassCreationHandle.createWithCustomLookup(MetaClassRegistry.java:156)
	at groovy.lang.MetaClassRegistry$MetaClassCreationHandle.create(MetaClassRegistry.java:139)
	at org.codehaus.groovy.reflection.ClassInfo.getMetaClassUnderLock(ClassInfo.java:272)
	at org.codehaus.groovy.reflection.ClassInfo.getMetaClass(ClassInfo.java:315)
	at org.codehaus.groovy.reflection.ClassInfo.getMetaClass(ClassInfo.java:325)
	at org.codehaus.groovy.runtime.metaclass.MetaClassRegistryImpl.getMetaClass(MetaClassRegistryImpl.java:262)
	at org.codehaus.groovy.runtime.InvokerHelper.getMetaClass(InvokerHelper.java:983)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.createPojoSite(CallSiteArray.java:116)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.createCallSite(CallSiteArray.java:160)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
	at com.example.MyExceptionHandler.handle(MyExceptionHandler.groovy:29)
```

I suspect Groovy is trying to find JacksonConfiguration when it tries to load the `HateoasErrorResponseProcessor`

### Steps To Reproduce

1. Clone linked project
2. Run project
3. curl http://localhost:8080

### Example Application

https://github.com/bloidonia/serde-groovy-issue

### Version

3.7.1

**Comment by timyates** [comment]
So adding `problem-json` to the guid application makes this go away as it uses that instead of the `HateoasErrorResponseProcessor` in core.

Maybe we need to add a `HateoasErrorResponseProcessor` here (in a separate lib?) to override the one in core if people pull in Serde?

**Comment by timyates** [comment]
So there is a fix available.  If we delete this constructor

https://github.com/micronaut-projects/micronaut-core/blob/eed72283006f4428aa7e6e82f9612433008bc1dc/http-server/src/main/java/io/micronaut/http/server/exceptions/response/HateoasErrorResponseProcessor.java#L49-L58

Then the attached Groovy example project works.

But this breaks binary compatibility

<img width="1147" alt="image" src="https://user-images.githubusercontent.com/49317/196461776-5fbdf117-439d-45d6-adfe-4ecfb240f712.png">

So are we ok with this fix going in for 4.0.0, and we will make that constructor as Deprecated on 3.7.x?

There is a workaround (the user can provide their own `HateoasErrorResponseProcessor` or switch to problem-json)

**Comment by timyates** [comment]
Deprecation for 3.8.0 here https://github.com/micronaut-projects/micronaut-core/pull/8185
Removal for 4.0.0 here https://github.com/micronaut-projects/micronaut-core/pull/8186
